xarray/core/variable.py

```diff
@@ -1076,10 +1101,16 @@ def __init__(self, dims, data, attrs=None, encoding=None, fastpath=False):
                              type(self).__name__)
 
     def _data_cached(self):
-        if not isinstance(self._data, PandasIndexAdapter):
-            self._data = PandasIndexAdapter(self._data)
+        # Unlike in Variable._data_cached, always eagerly resolve dask arrays
```

Comment:
I thought we wanted to always eagerly load this into memory without caching at all?

That would suggest we should put something like `self._data = PandasIndexAdapter(self._data)` in the constructor, and make `_data_cached` and `_data_cast` on the subclass dummy methods.